qldbdriver/result.go

```diff
@@ -66,9 +68,51 @@ func (result *Result) getNextPage() error {
 	result.pageValues = nextPage.Page.Values
 	result.pageToken = nextPage.Page.NextPageToken
 	result.index = 0
+	result.updateMetrics(nextPage)
 	return nil
 }
 
+func (result *Result) updateMetrics(fetchPageResult *qldbsession.FetchPageResult) {
+	if fetchPageResult.ConsumedIOs != nil {
+		if result.consumedIOs == nil {
```

Comment:
Why not set consumed IOs to zero on construction of a result, then you don't need this nil check

Reply:
Updated construction of `result` to set `consumedIOs`/`timingInformation` to 0. Removed this check.